storage/innobase/xtrabackup/test/inc/common.sh

```diff
@@ -477,25 +477,40 @@ function force_checkpoint()
 ########################################################################
 # Configure a specified server as a slave
 # Synopsis:
-#   setup_slave <slave_id> <master_id>
+#   setup_slave [GTID] <slave_id> <master_id> ... <master_id>
 #########################################################################
 function setup_slave()
 {
-    local slave_id=$1
-    local master_id=$2
+    local extra=""
+    if [ "$1" == "GTID" ]
+    then
+        extra=", MASTER_AUTO_POSITION = 1"
+        shift
+        echo "Setting up slave in GTID mode"
+    fi
 
-    vlog "Setting up server #$slave_id as a slave of server #$master_id"
+    local -r slave_id_arg=$1
+    shift
 
-    switch_server $slave_id
+    switch_server $slave_id_arg
+    while [ "$#" -ne 0 ]
+    do
+        local master_id_arg=$1
+        shift
 
-    run_cmd $MYSQL $MYSQL_ARGS <<EOF
+        vlog "Setting up server #$slave_id_arg as a slave of server #$master_id_arg"
+
+        run_cmd $MYSQL $MYSQL_ARGS <<EOF
 CHANGE MASTER TO
   MASTER_HOST='localhost',
   MASTER_USER='root',
-  MASTER_PORT=${SRV_MYSQLD_PORT[$master_id]};
-
-START SLAVE
+  MASTER_PORT=${SRV_MYSQLD_PORT[$master_id_arg]}
+  $extra
+  FOR CHANNEL 'master-$master_id_arg';
```

Comment:
Fixed, now one have to explicitly specify `USE_CHANNELS` as an argument to the `setup_slave` in order to get named channels in `CHANGE MASTER` statement.